src/domains/transaction/validations/UsernameRegistration.ts

```diff
@@ -1,10 +1,32 @@
 import { Networks } from "@ardenthq/sdk";
 import { ValidateResult } from "react-hook-form";
-import { validatePattern } from "@/utils/validations";
+import { MutableRefObject } from "react";
 import { debounceAsync } from "@/utils/debounce";
 
+const validateUsername = (t: any, value: string): string | undefined => {
```

Comment:
I could have a regex to check all of the cases below but in this case we wouldn't have a descriptive error message, that is why I pick this approach.

Reply:
spaces aren't valid either way, so `trim` shouldn't be needed as it will fail validation on the space itself